packages/design-system/src/components/dropDown/select/components.js

```diff
@@ -91,27 +91,30 @@ SelectButton.propTypes = {
   isOpen: PropTypes.bool,
 };
 
-export const ChevronWrap = styled.div(
-  ({ theme, isOpen }) => css`
-    color: ${theme.colors.fg.secondary};
-    width: 32px;
+const DisclosureIcon = styled(ChevronDownSmall)(
+  ({ theme }) => css`
     height: 32px;
+    width: auto;
+    color: ${theme.colors.fg.secondary};
+    transition: transform 100ms;
 
-    ${isOpen &&
-    css`
+    &.open {
       transform: rotate(180deg);
-    `}
+    }
   `
 );
-ChevronWrap.propTypes = {
+
+export const Disclosure = ({ className = '', isOpen = false, ...other }) => (
+  <DisclosureIcon
+    className={`${className} ${isOpen ? 'open' : ''}`.trim()}
```

Comment:
If we do decide to start using class names in this way, I think it makes sense to use the tiny utility library [`classnames`](https://www.npmjs.com/package/classnames) to clean things like this up:

```js
import cx from 'classnames';
...
  <DisclosureIcon className={cx(className, {open: isOpen})} {...other} />
```

It is probably worth a discussion.

Reply:
Looking at this component again, does it ever receive the `className` property? Is it used as `styled(Disclosure)` somewhere I'm not aware of?

Reply:
It does, thanks a lot for sharing your feedback & experience! It's much appreciated.

Definitely agree that there's room for improvement in this area, espcially for generic class names like "Container" or "Wrapper". Those are good examples of names that we see across the code base.

I've now opened #11749 to further discuss this and brainstorm on ideas.

Reply:
> Looking at this component again, does it ever receive the `className` property? Is it used as `styled(Disclosure)` somewhere I'm not aware of?

By modifying className dynamically, styling can be defined in a parent component, eliminating the need to call `styled(Disclosure)` and passing props to define styling. If a parent already has access to a prop used for styling, it doesn't need to be passed on to affected children. Hopefully I explained that properly. I can share a code snippet to show what I mean.

Reply:
> > Looking at this component again, does it ever receive the `className` property? Is it used as `styled(Disclosure)` somewhere I'm not aware of?
> 
> By modifying className dynamically, styling can be defined in a parent component, eliminating the need to call `styled(Disclosure)` and passing props to define styling. If a parent already has access to a prop used for styling, it doesn't need to be passed on to affected children. Hopefully I explained that properly. I can share a code snippet to show what I mean. One possible issue that comes to mind with this approach would be unintentional styling of child components that could have the className value applied. I think the natural CSS cascade would handle this, but there could be specificity issues that would interfere. 🤔

Yeah I know how it works. but [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) dictates that we should only add functionality we actually need. And if this component never needs restyling, it's just useless code. That's my point, sorry if that wasn't clear.

Reply:
I totally agree, it’s just a matter of where component-specific styling is defined - wrapped in `styled(AwesomeComponent)` or defined in a parent component with a selector to target `AwesomeComponent`. This is used elsewhere in the codebase, but with element or pseudo selectors.